tests/functional/ArchiveTest.php

```diff
@@ -123,6 +123,29 @@ public function testArchiveDumpCommand(): void
         );
     }
 
+    public function testArchiveDumpSymlinkSwapCommand(): void
+    {
+        // Sites that contain symlinks to files outside the project root cause
+        // critical errors. To test for this, we manually create a symlink
+        // and then run archive:dump. If it completes at all, the symlink fix
+        // from Issue #5991 is working.
+        $linktarget      = Path::join("/tmp", 'symlinktest.txt');
+        $linkdestination = Path::join($this->getSandbox(), 'symlinkdest.txt');
+
+        file_put_contents($linktarget, "This is a symlink target file.");
+        symlink($linktarget, $linkdestination);
```

Comment:
We write to the sandbox and not hard code /tmp. Furthermore we register files for cleanup. This needs cleanup upon success or failure.